src/archive_librarian/dbHandler.h

```diff
@@ -0,0 +1,137 @@
+                           FileSet& epochHistoryFileSet_,
+                           StatusData& statusData_);
+    
+
+    std::pair<int,int> jobIdLookup(int clusterId, int procId);
+    void updateFileInfo(FileInfo epochHistoryFile, FileInfo historyFile);
+
+    // Helper functions for archiveMonitor and status
+    FileInfo lastFileRead(const std::string& type); // To get the last file read from the Status table
+    int countUnprocessedFiles(); // Counts files with 0 lastOffset 
+    bool insertNewFiles(const std::pair<int, std::string>& rotatedFile, std::vector<FileInfo>& newFiles);
+    bool deleteFiles(const std::vector<int>& deletedFileIds);
+    bool insertEpochFileRecords(const std::vector<EpochRecord>& records, const FileInfo& fileInfo);
+    bool insertJobFileRecords(const std::vector<JobRecord>& jobs, const FileInfo& fileInfo);
+    
+
+
+    // To read EpochHistory file with all record types
+    bool insertUnseenJob(const std::string& owner, int clusterId, int procId, int64_t timeOfCreation);
+    void batchInsertEpochRecords(const std::vector<EpochRecord>& records);
+
+    // User query functions
+    std::vector<std::tuple<int, int, int>> getJobsForUser(const std::string& username);
+    std::vector<std::pair<std::string, int>> getJobCountsPerUser();
+
+    // Helper function for getting the database for dbHandlerTestUtils
+    sqlite3* getDB() const { return db_; }
+
+private:
+
+    class JobIdCache {
+    public:
+        // Composite key type for (clusterId, procId)
+        struct Key {
+            int clusterId;
+            int procId;
+            
+            bool operator<(const Key& other) const {
+                if (clusterId != other.clusterId) {
+                    return clusterId < other.clusterId;
+                }
+                return procId < other.procId;
+            }
+            
+            bool operator==(const Key& other) const {
+                return clusterId == other.clusterId && procId == other.procId;
+            }
+        };
+        
+        // Value type storing jobId, jobListId, and timestamp
+        struct Value {
+            int jobId;
+            int jobListId;
+            long timestamp;
+            
+            Value() : jobId(-1), jobListId(-1), timestamp(0) {}
+            Value(int jId, int jListId, long ts) : jobId(jId), jobListId(jListId), timestamp(ts) {}
+        };
+
+        explicit JobIdCache(size_t maxSize = 10000);
+        ~JobIdCache() = default;  
+
+        std::pair<int,int> get(int clusterId, int procId);
+        void put(int clusterId, int procId, int jobId, int jobListId, long timestamp); 
+        void updateTimestamp(int clusterId, int procId, long timestamp);  
+        bool exists(int clusterId, int procId) const;
+        void remove(int clusterId, int procId);
+        size_t size() const;
+        void clear();
+
+
+    private:
+        Cache<Key, Value> cache_;  // Replace all the old private members with this
+    };
```

Comment:
Discussed in person, but lets replace the wrapping `JobIdCache` class completely with the use a `Cache` variable. Note: The Key/Value structs need to exist still and should be renamed to something more specific than `Key` and `Value`